assertk/src/commonTest/kotlin/test/assertk/AssertFailureTest.kt

```diff
@@ -24,6 +21,30 @@ class AssertFailureTest {
         assertSame(expected, assertFailure { throw expected }.valueOrFail)
     }
 
+    @Test
+    fun failure_with_expected_type_and_additional_properties() {
+        assertFailureWith<DummyException> {
+            throw DummyException(12, "night")
+        }.hasProperties(
+            DummyException::value to 12,
+            DummyException::tag to "night"
+        )
```

Comment:
In general the library does not use its own assertion mechanism to test itself since that could lead to bugs not getting caught by the tests.

For this case, however, I don't think it's worthwhile asserting at such a granular level. The only thing we care about here is that the caught exception is the same as the one thrown, so that's all we need to test (the above test also does this pattern):
```suggestion
        val expected = throw DummyException(12, "night")
        val actual = assertFailureWith<DummyException> { throw expected }
        assertSame(expected, actual.valueOrFail)
```

Reply:
I did this and put the tests for `hasProperties` in a better place... I think 😆